sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -948,6 +948,12 @@ internal virtual void SetNameFieldsIfNull()
         /// <see cref="CreateIfNotExists"/>
         /// instead.
         /// </remarks>
+        /// <remarks>
+        /// Note that this method does not guarantee that the path type
+        /// (file/directory) matches expectations.  For example, a
+        /// DataLakeFileClient representing a path to a directory
+        /// will return true, and vice versa.
+        /// </remarks>
```

Comment:
I have to say I don't understand what this doc comment is trying to convey. Could you try to reword it?

Reply:
Fixed.